models/ed/R/model2netcdf.ED2.R

```diff
@@ -643,11 +647,25 @@ put_T_values <- function(yr, nc_var, out, lat, lon, begins, ends, ...){
   
   
   # ----- define ncdf dimensions
-  
+  # setup output time and time bounds
+  sub_date_range <- seq(begins, by = "day", length.out = length(lubridate::yday(begins):ends))
+  sub_jdates <- lubridate::yday(lubridate::floor_date(seq(sub_date_range[1], sub_date_range[length(sub_date_range)],length.out= length(out[[1]]))))
+  iter_per_day <- length(sub_jdates[sub_jdates==lubridate::yday(sub_date_range[2])])  # look at day 2 to determine iterations per day (i.e. model timestep)
+  #iter_per_day <- length(out[[1]])/ends    ### does not work for part of a year!!
+  rm(sub_jdates)
+  step <- seq(0, 0.99, 1/iter_per_day)
+  sub_dates <- rep(sub_date_range,each=iter_per_day)
+  jdates <- lubridate::yday(sub_dates)
+  day_repeats <- as.vector(base::table(jdates)) # number of model outputs per day, e.g. 48 for half-hourly
+  tvals <- (jdates+step)-1
+  bounds <- array(data=NA, dim=c(length(tvals),2))
+  bounds[,1] <- tvals; bounds[,2] <- bounds[,1]+(1/iter_per_day); bounds <- round(bounds,4)  # create time bounds for each timestep in t, t+1; t+1, t+2... format
```

Comment:
I can see why you put these on the same line, but I would spread them out on separate lines for readability.

Also, for the entire file, assuming you're editing this in Rstudio, I would turn on its built in code linter to the strictest setting and follow its recommendations -- there are a lot of operator spacing, etc. style things in here. If you'd like, I can fix the style myself tomorrow -- should only take me a few minutes.

Reply:
@ashiklom I *think* I have done this but not certain, nor how to get Rstudio to show me issues it suggests I "fix". Looking for a readme on how to set that up

Reply:
Ran this

```
install.packages('lintr')
library(lintr)

```

then looks like running lintr::lint() on the script.  I could not see how after enabling Show Diagnostics in Rstudio to get it to run a check on the script...I even clicked the magic wand/Show Diagnostics but it didnt return anything :/

Reply:
Huh, weird. I was going to point you to [this](https://community.rstudio.com/t/lintr-integration-with-rstudio/1807/5), particularly the "Style diagnostics", but it sounds like you already did that. Yeah, `lintr` is what I usually use -- it has very good integration with automated Emacs ESS + flycheck (what I currently use) or `vim + Nvim-R + ale`. I haven't used RStudio in ages.

Reply:
@ashiklom I am getting a lot of these:

![screen shot 2018-09-11 at 9 44 57 am](https://user-images.githubusercontent.com/3067369/45364239-a2c84580-b5a7-11e8-8e96-dcaf5fb0240d.png)

I dont really know what it wants me to do as I dont see any trailing whitespace??

Reply:
I mean, I don't think it's absolutely essential to heed every single one of its comments, so use your judgement. That's why I like manual changes based on linter recommendations much more than automatic changes from e.g. `styler` -- I think every programming style guide has a rule like "If you have to choose between this style guide and having readable code, break the rules of the style guide."

FWIW, I did find some trailing whitespace in the beginning of this file, not exactly on those lines, but close, so the file has probably changed with more recent commits. But that's not a huge issue -- I was mostly commenting about using `=` instead of `<-` for assignment, leaving spaces around infix operators (e.g. `function(var = value)` and `x + y / z` rather than `function(var=value)` and `x+y/z`), which the linter is really good at picking up.